docs: remove references to hambach testnet (#1779)
regen-network/regen-ledger: docs/ledger/get-started/README.md

## Building From Source

> The following installation instructions includes one option for installing the `regen` binary with the "stable app configuration".

Reviewer: ```suggestion
The following installation instructions are for installing the `regen` binary.
```

> The following installation instructions include two options, one for installing the `regen` binary with the "stable app configuration" and another with the "experimental app configuration".

Reviewer: This is not true anymore. Can you update?

git checkout v5.0.0
```

Build and install the `regen` binary:

> *For the stable app configuration (used on Regen Mainnet and Redwood Testnet):*

Reviewer: This doesn't seem necessary anymore. Can you remove throughout the docs?

You should see the following:

```bash

> v5.0.0 

Reviewer: ```suggestion
v5.0.0
```
